dotnet/src/Connectors/Connectors.Memory.AzureCognitiveSearch/AzureCognitiveSearchMemoryStore.cs

```diff
@@ -198,13 +198,14 @@ public async IAsyncEnumerable<string> UpsertBatchAsync(string collectionName, IE
 
         if (searchResult == null) { yield break; }
 
+        var minAzureSearchScore = CosineSimilarityToScore(minRelevanceScore);
         await foreach (SearchResult<AzureCognitiveSearchMemoryRecord>? doc in searchResult.Value.GetResultsAsync())
         {
-            if (doc == null || doc.Score < minRelevanceScore) { continue; }
+            if (doc == null || doc.Score < minAzureSearchScore) { continue; }
 
             MemoryRecord memoryRecord = doc.Document.ToMemoryRecord(withEmbeddings);
 
-            yield return (memoryRecord, doc.Score ?? 0);
+            yield return (memoryRecord, ScoreToCosineSimilarity(doc.Score ?? 0));
```

Comment:
```suggestion
            yield return (memoryRecord, ScoreToCosineSimilarity(doc.Score ?? 0.5));
```
Sounds like `score: [0.5,1.0]` so probably should use that here.

Reply:
actually the minimum is 1/3, I'll add the check to the function and leave it to 0 here